### DISCUSSION TITLE
Triggering an email when a row is created

### DISCUSSION THREAD

**Discussion by KernOil** [Budibase/budibase#2016] [Help] [answered]
I am having some problems with the automation tab. I have it set up to email me the content of a new row when it is created. the problem I am having is it just returns [object Object] in the email instead of the new line. is there something I can do to fix this?

here is my email responce 
![image](https://user-images.githubusercontent.com/80058238/126005781-93115a3f-7fe2-4d03-9de9-2777115e63c2.png)

here is my current setup using {{trigger.row}}
![image](https://user-images.githubusercontent.com/80058238/126005834-e790cb36-aac8-4ef3-8f17-c9961a84e929.png)

this is using the internal DB, there are 2 text fields and one attachment field, below is the form I am submitting.
![image](https://user-images.githubusercontent.com/80058238/126005951-3c8b0c5b-4e9f-44d0-8108-0bd4e3c47d16.png)

any help would be great, Thank you

**Comment by mike12345567** [comment, marked as answer]
Hi @AndrewJRamay, you either need to write out each individual property as you want to see it like `{{ trigger.row.name }}` and `{{ trigger.row.description }}` or you can print it with a JSON stringify, like `{{ JSONstringify trigger.row }}`.

It is also worth noting that right now attachments cannot be sent in emails, this is something we have not been able to support as of yet.

**Reply by KernOil** [reply]
Thank you @mike12345567 I was just using the preloaded suggestions makes sense it needs a second .____ to point it at the correct object.

**Reply by GuilhermeHirsch** [reply]
Is it possible to use bracket notation, @mike12345567? For example: {{ trigger.row[ 'MÊS DE COMPETÊNCIA' ] }}.

**Reply by mike12345567** [reply]
Hi @GuilhermeHirsch - brackets have a slightly different meaning in Handlebars, generally they mean this is a literal string, don't treat it as anything else.

What you have there is very close to what you want, I created the scenario myself to test it, but essentially everything inside the brackets gets used, so you don't want the spaces or the quotes.

This worked for me:
![image](https://user-images.githubusercontent.com/4407001/126645511-fbd3e91b-df3b-434b-ac62-9aed297ccc48.png)

Its a little bit different to the way that Javascript would represent this, but its just due to the fact the square brackets essentially mean "everything inside here is the exact key I want".

**Reply by GuilhermeHirsch** [reply]
Excellent, @mike12345567! Understood, thank you very much for the support.

**Reply by KernOil** [reply]
Hi Mike, 
I have been having trouble getting this to work with the SQL-connected database. 
I would like to have an email sent when a new row is added, so I have set up the automation like I did before (above) calling a column in the new row, and I keep getting no content returned.
![image](https://user-images.githubusercontent.com/80058238/131741757-c0874887-02f4-4295-8c85-bff926208f2f.png)

If I try to print the entire row with {{ JSONstringify trigger.row }} I will get the string ID
![image](https://user-images.githubusercontent.com/80058238/131741337-10bc9723-83a6-4f88-bbf9-09ca3fb9faea.png)

can you suggest anything else I can try?

**Reply by mike12345567** [reply]
Hi @AndrewJRamay - just to confirm, you are trying to trigger an automation when you write a row to a SQL DB from Budibase?

Right now this isn't fully implemented/tested as there is no way for Budibase to detect when something externally writes a row to the database, which seems a little confusing as it'd only work when used with Budibase. None the less however it should probably still work when writing rows exclusively from Budibase.

**Reply by KernOil** [reply]
@mike12345567  Yes that is correct submitting a row in BB to an external SQL DB. I figured it should work just the same as if it were an internal Db but for whatever reason, I keep getting the no content.

what my end goal is, I would like to have an email sent to the person that the task is assigned to. The send-to portion of the automation would have to read the email address from the row of the name that the FK is associated with, from the table that the form is being saved to.  is that possible / do you understand my description? 
here is a picture of my description.
![image](https://user-images.githubusercontent.com/80058238/131746086-65dd0f58-7a81-4532-826e-b42d758de961.png)
